### PR TITLE
Remove NuGet.Packaging.Core usage from arcade, use NuGet.Packaging instead

### PR DESCRIPTION
**Pull request by nkolev92**
Fixes https://github.com/dotnet/arcade/issues/12906

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

File: src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs [hunk 1]
```diff
@@ -33,18 +33,19 @@ public override SignatureVerificationResult VerifySignature(string path, string
 
         private bool IsSigned(string path)
         {
-            IEnumerable<ISignatureVerificationProvider> providers = SignatureVerificationProviderFactory.GetSignatureVerificationProviders();
```
Comment (nkolev92) [inline src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs:36]: This is the only change that actually changes behavior. 

I'm unsure if this is still being used, but unlikely that this is related to the problems the .NET SDK encountered, since a lot of the types have changed. Example: SignatureVerificationProviderFactory does not exist in 4.9. 

I'd be happy to revert this change if you prefer to minimize the risk.

Comment (nkolev92) [reply src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs:36]: cc @dtivel 

Can you please take a look at this change?

The project is .NET Framework and likely only runs on windows. 

https://github.com/NuGet/NuGet.Client/blob/16d8eb328c50cbc1c0a2063578745562e6a24792/src/NuGet.Core/NuGet.Packaging/Signing/SignatureVerificationProviderFactory.cs went away in 4.9.

**Review by dtivel — APPROVED** [review]

**Comment by nkolev92** [comment]
cc @riarenas @mmitche can you help review this?

Thanks.

File: src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs (hunk 1 above)
Comment (mmitche) [reply src/SignCheck/Microsoft.SignCheck/Verification/NupkgVerifier.cs:36]: This does run, but only on Windows and .NET Framework.

**Review by mmitche — APPROVED** [review]
